src/counts.jl

```diff
@@ -237,13 +237,62 @@ Add counts based on `x` to a count map. New entries will be added if new values
 If a weighting vector `wv` is specified, the sum of the weights is used rather than the
 raw counts.
 """
-function addcounts!(cm::Dict{T}, x::AbstractArray{T}) where T
+function addcounts!(cm::Dict{T}, x::AbstractArray{T}; uselessmem = false) where T
```

Comment:
I don't like arguments like `uselessmem`. Who would want to use more memory? Better not have an argument for now, we can always add it later if we find cases where it's really useful.

Reply:
The radixsort method is faster but uses more memory than before. Maybe just a better name, Perhaps `usedict` so that it uses the dictionary method and explain in the notes that it uses less memory.

Reply:
Ah, of course. Maybe we could have `alg = :auto`, and accept `:dict` and `:radixsort`? Would be nice to document the amount of memory used by radix sort (i.e. it makes a copy before sorting).